Only capitalize if no uppercase letters

wasimsandhu opened the pull request:
## Summary

For all capitalization methods in `apc.mpl`, only capitalize text when text has no uppercase letters. This fixes the issue of incorrectly changing text like "PC1" to "Pc1", or "mRNA" to "MRNA".

As a driveby change, performs the same change to the `capitalize_ticklabels` methods in `apc.plotly` (I appeared to have missed these in #73).

Addresses #55.

## PR checklist

- [x] Tag the issue(s) or milestones this PR fixes (e.g. `Fixes #123, Resolves #456`).
- [x] Describe the changes you've made.
- [x] Describe any tests you have conducted to confirm that your changes behave as expected.
- [x] If you've added new software dependencies, make sure that those dependencies are included in the appropriate `conda` environments.
- [x] If you've added new functionality, make sure that the documentation is updated accordingly.
- [x] If you encountered bugs or features that you won't address, but should be addressed eventually, create new issues for them.

keithchev reviewed (changes requested):
I'm not sure this is quite right because `isupper()` returns true only if *all* characters are uppercase, so this will fix the "PC1" -> "Pc1" problem but not the "mRNA" -> "MRNA" problem. I think it would be more robust to check if *any* letters in the label are capitals, and then only call `capitalize` if there are no capital letters in the label. FWIW, this is also one of the approaches suggested in #55.

wasimsandhu commented:
> I think it would be more robust to check if any letters in the label are capitals, and then only call capitalize if there are no capital letters in the label. FWIW, this is also one of the approaches suggested in https://github.com/Arcadia-Science/arcadia-pycolor/issues/55.

@keithchev Ah yeah, this is what I intended to do, but `isupper` was a mistake. Changed all instances of `not {text}.isupper()` to `{text}.islower()` to check if all characters in a string are lowercased. Also updated docstrings. Appears to be working as intended with both examples I offered.

keithchev reviewed (approved):
Just a side note that at some point it might be worth de-duplicating this capitalization logic, simple as it is, since it is currently duplicated in 19(!) different places.
